xpk.py

```diff
@@ -1839,6 +1892,38 @@ def setup_docker_image(args) -> tuple[int, str]:
   return 0, docker_image
 
 
+def get_device_and_accelerator_type(args) -> tuple[str, int]:
+  device_type = args.tpu_type if args.tpu_type else args.device_type
+  # check if device_type is in TPU list
+  accelerator_type = AcceleratorType['TPU'] \
+    if device_type in TpuUserFacingNameToSystemCharacteristics \
+      else AcceleratorType['GPU']
+  return device_type, accelerator_type
```

Comment:
What if the device_type is not GPU or TPU?

Reply:
I have a few suggestions here:
1. Combine the TPU and GPU user facing characteristics dictionary. We store the AcceleratorType within the system_characteristics. 
2. Add device_type to system_characteristics or pass it as an argument
3. Combine the two functions `get_device_and_accelerator_type` and `get_system_characgeristixs`